src/PdfSharper/Pdf/PdfDocument.cs

```diff
@@ -799,6 +799,9 @@ private PdfReference[] GatherFirstPageAndDocumentReferences()
 
             var firstPageObjects = PdfTraversalUtility.TransitiveClosure(Pages[0], exclusions).Select(kvp => kvp.Key).ToList();
 
+            //sometimes the transitiveclosure does not include the root element
+            firstPageObjects.Add(Pages[0].Reference);
```

Comment:
because of the "sometimes", should we make sure it isn't there before adding - prevent duplicate entries?